src/app/domain/backup/services/uploadMedias.ts

```diff
@@ -51,7 +51,18 @@ export const uploadMedias = async (
     }
 
     try {
-      const uploadUrl = getUploadUrl(client, backupFolderId, mediaToUpload)
+      const uploadFolderId = await getUploadFolderId(
+        client,
+        localBackupConfig,
+        mediaToUpload
+      )
```

Comment:
An async call inside a loop, I don't like it. 

nit: Also this method should be rename to something like getOrCreateFolder

Reply:
I keep this review for a next PR about optimization.